src/types.ts

```diff
@@ -1219,14 +1219,19 @@ export interface PublicKeyInfoArgs {
     compressed?: boolean;
 }
 
-export interface PublicKeyInfoForVaultAccountArgs {
+export interface PublicKeyInfoByAccountAssetArgs {
     assetId: string;
-    vaultAccountId: number;
+    accountId: number;
     change: number;
     addressIndex: number;
     compressed?: boolean;
 }
 
+export interface PublicKeyInfoForVaultAccountArgs
+    extends Omit<PublicKeyInfoByAccountAssetArgs, "accountId"> {
+    vaultAccountId: number;
+}
+
```

Comment:
omitting accountId for vaultAccoutId, where the base type is `PublicKeyInfoByAccountAssetArgs` is weird.

i think its better to either create a base interface, where ncw\vaults extends it, with the correct fields, or just use one interface, and change the names of the params as needed in each function

Reply:
this is due to backward compatability

Reply:
there are no backward issues with my suggestion, it's just a better typing style

Reply:
something like:

```
export interface BasePublicKeyInfoByAccountAssetArgs {
    assetId: string;
    change: number;
    addressIndex: number;
    compressed?: boolean;
}
export interface PublicKeyInfoForVaultAccountArgs extends BasePublicKeyInfoByAccountAssetArgs {
    vaultAccountId: number;
}
export interface PublicKeyInfoForNCWAccountArgs extends BasePublicKeyInfoByAccountAssetArgs {
    accountId: number;
}
```